### DISCUSSION TITLE
How to add header border ?

### DISCUSSION THREAD

**Discussion by ardyfeb** [react-navigation/react-navigation#9359] [Q&A] [answered]
**Current Behavior**
I currently migrating to react-navigation 5 from react-navigation 4, there so many missing option that version 4 has

I cannot add border style and another view style to header. Current options available in `headerStyle` is 
```ts
headerStyle?: {
    backgroundColor?: string;
    blurEffect?: ScreenStackHeaderConfigProps['blurEffect'];
};
```

**Expected Behavior**

Header style should accept view style, user should freely add own style to header. Why react-navigation restrict this with no reason ?

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | Android
| @react-navigation/native       | ^5.9.2
| @react-navigation/stack        | ^5.14.2

**Comment by rawatnaresh** [comment, marked as answer]
As the [docs](https://reactnavigation.org/docs/headers#adjusting-header-styles) says 

> headerStyle: a style object that will be applied to the View that wraps the header

which means you can pass any valid style props.
```js
headerStyle: {
  backgroundColor: '#f4511e',
  borderBottomColor: 'yellow',
  borderBottomWidth: 5
},
```

![Screen Shot 2021-02-25 at 09 18 51](https://user-images.githubusercontent.com/29520305/109099063-848c3b00-774a-11eb-9741-f86cb1033488.png)

**Reply by syedumeraliSalsoft3552** [reply]
this is not working either on react-navigation 6

**Reply by raajnadar** [reply]
It will work for react navigation header and not for custom header. If that is not working add a snack repro.

**Reply by adherb** [reply]
This isn't working on Version 6, does anyone know how to fix this?

**Reply by raajnadar** [reply]
You need to use https://reactnavigation.org/docs/material-top-tab-navigator/#tabbarindicatorstyle for react-navigation v6

**Reply by sospedra** [reply]
I ran into the same issue. In my case, I was not using `stack` but `native-stack`. The [former](https://github.com/react-navigation/react-navigation/blob/4caf3cb849a8708a36d0dbf22dac432d2ca780be/packages/elements/src/types.tsx#L166-L169) accepts any StyleProp while the [latter](https://github.com/react-navigation/react-navigation/blob/4caf3cb849a8708a36d0dbf22dac432d2ca780be/packages/native-stack/src/types.tsx#L243-L249) only `backgroundColor`.

I fixed it by [styling](https://stackoverflow.com/a/74830694/2824333) the `borderTop` of the `contentStyle`.
